runtime/cudaq/spin/spin_op.cpp

```diff
@@ -398,7 +400,9 @@ spin_op &spin_op::operator*=(const spin_op &v) noexcept {
       theirRow++;
   }
 
+#ifdef CUDAQ_HAS_OPENMP
```

Comment:
Does this do anything? I thought that if you did not build with `-fopenmp` then all OpenMP pragmas are ignored.

Reply:
If I commented out the `find_package(OpenMP)` calls our `CMakeLists.txt` files, this file would fail to compile because it was an unrecognized pragma. (It was probably a warning, but we fail on nearly all warnings.)

Reply:
Here is the full error message:
```bash
/workspaces/cuda-quantum/runtime/cudaq/spin/spin_op.cpp:401: error: ignoring ‘#pragma omp parallel’ [-Werror=unknown-pragmas]
  401 | #pragma omp parallel for shared(composition)
      | 
cc1plus: all warnings being treated as errors
```